src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/DocumentFilters/MemberSummaryToSchemaDescriptionFilter.cs

```diff
@@ -58,23 +59,18 @@ public class MemberSummaryToSchemaDescriptionFilter : IDocumentFilter
                              m.Attribute(KnownXmlStrings.Name).Value.StartsWith("P:")));
             }
 
-            foreach (var propertyMember in propertyMembers)
-            {
-                var fullPropertyName = propertyMember.Attribute(KnownXmlStrings.Name).Value;
-
-                var splitPropertyName = fullPropertyName.Split('.');
-
-                // Take everything before the last period and remove the "P:" prefix.
-                var className =
-                    string.Join(".", splitPropertyName.Take(splitPropertyName.Length - 1))
-                        .Substring(startIndex: 2);
+            var typeNames = propertyMembers.Attributes(KnownXmlStrings.Name).Select(i => string
+                .Join(".", i.Value.Split('.').Take(i.Value.Split('.').Length - 1))
+                .Substring(2)).Distinct().ToList();
```

Comment:
This is very hard to read and understand. Lots of parens and nesting going on.

Can we do any of the following or a mix to make it better? Feels free to use your judgement.
1. Break it up by adding more intermediate local variable assignments, which would help describe what is going on better via the variable names, or (2) add comments to describe the logic, or (3) use newlines and indentation to make the parts more readable.

Reply:
sure will do 2) and 3)

---
In reply to: [218900909](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/154#discussion_r218900909) [](ancestors = 218900909)